### ISSUE TITLE
Directly open photo in fullscreen mode after thumbnail click

### ISSUE THREAD

**Issue by whuthwhut123** [dimsemenov/PhotoSwipe#1422]
When a user clicks an image in my the thumbnail gallery i want it to open direcly in fullscreen mode.
Normally a user has to click on the button in the bar (see picture), I want to skip this step.
Any suggestions on how to achieve this?
![image](https://user-images.githubusercontent.com/5947133/30366089-32a462c6-986a-11e7-9543-6f1cfc5ba389.png)

**Comment by acwolff** [comment]
This is a standard feature of [my PhotoSwipe skin ](http://jalbum.net/en/skins/skin/PhotoSwipe), selectable by the user via the setup button in the upper right corner of the thumbnails page, see [this example album](http://andrewolff.jalbum.net/PS_varThumbsRowFixedRatio/).

I use next code:
`	  if (PC||Android) {
		indexPageFullScreen= screenfull.isFullscreen;
    	if (slidePageFullScreen) {
    		if (!screenfull.isFullscreen) screenfull.request();
		}
	  }
	  openPhotoSwipe(ind);
	}`
see also https://github.com/sindresorhus/screenfull.js/

**Comment by jp-onsite** [comment]
Your implementation have the same issues as mine: If you use chrome and have the chrome window not maximized as you click on the thumbnails, than the pictures are not zoomed the same way as if you start fullscreen on an already maximized window. This lead to more problems of overlapping pictures and issues with zoom. Another problem is that you have to hit the ESC key to close fullscreen twice: 1 time for fullscreen to exit and another time to exit photoswipe. If you click on the X in the upper right corner it will close fullscreen and photoswipe at the same time.

a quick fix is to add the function gallery.updateSize(true) after the initialsation of the gallery:
`setTimeout(function(){gallery.updateSize(true);}, 1000);`

**Comment by acwolff** [comment]
@jp-onsite thanks for your solution, I fid test it and it works, see this album:
https://andrewolff.jalbum.net/Vecht_SaharaN_PST/
But it does not solve the minor problem with the ESC key, it that correct?

**Comment by jp-onsite** [comment]
yes correct, the ESC key problem persist. Perhaps it's possible to trigger the close() event if the key is pressed but i havn't tried to solve this problem yet.
You can try to lower the timeout of 1000 but this is a pretty save time to avoid the error.
